rmw_connext_shared_cpp/src/shared_functions.cpp

```diff
@@ -319,6 +323,10 @@ create_node(const char * implementation_identifier, const char * name, size_t do
   }
   participant_qos.participant_name.name =
     static_cast<char *>(rmw_allocate((strlen(name) + 1) * sizeof(char)));
+  if (!participant_qos.participant_name.name) {
+    RMW_SET_ERROR_MSG("failed to allocate memory");
+    return NULL;
+  }
   participant_qos.participant_name.name = strdup(name);
```

Comment:
Yes... I was just adding the check that I thought was missing. But the malloc looks completely unnecessary and leaked because we use `strdup` there.